src/plugins/data/server/search/strategies/ese_search/ese_search_strategy.ts

```diff
@@ -134,6 +139,9 @@ export const enhancedEsSearchStrategyProvider = (
       const response = esResponse.body as estypes.SearchResponse<any>;
       return {
         rawResponse: shimHitsTotal(response, options),
+        ...(esResponse.meta?.request?.params
+          ? { requestParams: sanitizeRequestParams(esResponse.meta?.request?.params) }
```

Comment:
This assumes we always want the request params being sent in the response - this will cause inflated responses most of the time even in cases where it's not used (e.g. the response from a GET request after the search has already been submitted). Can we make this so the behavior is opt-in, or at least remove these parameters from cases where the client should already have the request parameters?

Reply:
How would this look? Would consumers of SearchSource pass in an option to turn-on request parameters?

Reply:
I updated server side logic to avoid returning requestParams for async polling requests 3f1bc365ed3b95db7f1ae12a83e9f4f506713b5e